### DISCUSSION TITLE
How can I stream collections without using deprecated method $$.stream()?

### DISCUSSION THREAD

**Discussion by balantain** [selenide/selenide#1772] [Q&A] [answered]
Hi, first of all I would like to give thanks to you for such a useful tool!
I'm looking for an opportunity to create a children ElementsCollection based on a parent's one. For example, I have a web table and a parent ElementsCollection consisting of table rows. So, after filtering this collection by some condition, I get, for example, 50 result rows. Then need to save the first cell in each row as a SelenideElement in a new ElementsCollection (children). This case doesn't have any issues, if I use List, because I can do this using stream() as:

```
List<SelenideElement> parents = $$("parent_css_selector");
List<SelenideElement> children = parents.stream().filter(s -> s.getText().equals("some_text")).map(s -> s.$("child_css_locator")).collect(Collectors.toList());

//or even in List<String> if I need to...

List<String> childrenTexts = parents.stream().filter(s -> s.getText().equals("some_text")).map(s -> s.$("child_css_locator")).getText().collect(Collectors.toList());
```
But since stream() was deprecated in Selenide 6.2.0 I cant find the most acceptable solution for this.

**Comment by asolntsev** [comment, marked as answer]
I recommend to avoid such long iterations etc. It causes slow tests.
Instead, I would write a proper xpath that finds all the needed elements with just one we driver call. 

I registered a feature request for adding non-deprecated `stream()` method: https://github.com/selenide/selenide/issues/1773

I really DON’T RECOMMEND using iterating elements this way. Such a test is highly ineffective. Just write a CollectionCondition as the deprecation notice recommends.
